Build error when Navigation 2D is disabled

### Tested versions

- Reproducible: master [f5918a9d35350bf6402dd1b4902ab539747d77a6]
- Not reproducible: 4.5.1.stable

### System information

Godot v4.6.dev (f5918a9d3) - Arch Linux #1 SMP PREEMPT_DYNAMIC Mon, 24 Nov 2025 15:21:09 +0000 on X11 - X11 display driver, Multi-window, 2 monitors - OpenGL 3 (Compatibility) - NVIDIA GeForce RTX 3060 Ti (nvidia; 580.105.08) - 13th Gen Intel(R) Core(TM) i5-13400F (16 threads) - 15.37 GiB memory

### Issue description

Building the engine fails when Navigation 2D is disabled.

<details><summary>Output</summary>

```
scons: Reading SConscript files ...
Automatically detected platform: linuxbsd
Building for platform "linuxbsd", architecture "x86_64", target "template_debug".
scons: done reading SConscript files.
scons: Building targets ...
[ 11%] Compiling scene/2d/cpu_particles_2d.cpp ...
[ 11%] Compiling scene/2d/polygon_2d.cpp ...
[ 11%] Compiling scene/2d/skeleton_2d.cpp ...
[ 11%] Compiling scene/2d/tile_map.cpp ...
[ 11%] Compiling scene/2d/tile_map_layer.cpp ...
[ 11%] Compiling scene/2d/visible_on_screen_notifier_2d.cpp ...
[ 11%] Compiling scene/2d/physics/area_2d.cpp ...
[ 11%] Compiling scene/2d/physics/character_body_2d.cpp ...
[ 11%] Compiling scene/2d/physics/collision_object_2d.cpp ...
[ 11%] Compiling scene/2d/physics/collision_polygon_2d.cpp ...
[ 11%] Compiling scene/2d/physics/collision_shape_2d.cpp ...
[ 11%] Compiling scene/2d/physics/kinematic_collision_2d.cpp ...
[ 11%] Compiling scene/2d/physics/physical_bone_2d.cpp ...
[ 11%] Compiling scene/2d/physics/physics_body_2d.cpp ...
scene/2d/tile_map_layer.cpp: In member function 'void TileMapLayer::_physics_draw_quadrant_debug(const RID&, DebugQuadrant&)':
scene/2d/tile_map_layer.cpp:1269:62: error: incomplete type 'Mesh' used in nested name specifier
 1269 |                                 face_mesh_array.resize(Mesh::ARRAY_MAX);
      |                                                              ^~~~~~~~~
scene/2d/tile_map_layer.cpp:1270:55: error: incomplete type 'Mesh' used in nested name specifier
 1270 |                                 face_mesh_array[Mesh::ARRAY_VERTEX] = Vector<Vector2>(face_vertex_array);
      |                                                       ^~~~~~~~~~~~
scene/2d/tile_map_layer.cpp:1271:55: error: incomplete type 'Mesh' used in nested name specifier
 1271 |                                 face_mesh_array[Mesh::ARRAY_INDEX] = Vector<int32_t>(face_index_array);
      |                                                       ^~~~~~~~~~~
scene/2d/tile_map_layer.cpp:1272:55: error: incomplete type 'Mesh' used in nested name specifier
 1272 |                                 face_mesh_array[Mesh::ARRAY_COLOR] = Vector<Color>(face_color_array);
      |                                                       ^~~~~~~~~~~
scene/2d/tile_map_layer.cpp:1276:62: error: incomplete type 'Mesh' used in nested name specifier
 1276 |                                 line_mesh_array.resize(Mesh::ARRAY_MAX);
      |                                                              ^~~~~~~~~
scene/2d/tile_map_layer.cpp:1277:55: error: incomplete type 'Mesh' used in nested name specifier
 1277 |                                 line_mesh_array[Mesh::ARRAY_VERTEX] = Vector<Vector2>(line_vertex_array);
      |                                                       ^~~~~~~~~~~~
scene/2d/tile_map_layer.cpp:1278:55: error: incomplete type 'Mesh' used in nested name specifier
 1278 |                                 line_mesh_array[Mesh::ARRAY_COLOR] = Vector<Color>(line_color_array);
      |                                                       ^~~~~~~~~~~
[ 11%] Compiling scene/2d/physics/ray_cast_2d.cpp ...
[ 11%] Compiling scene/2d/physics/rigid_body_2d.cpp ...
[ 11%] Compiling scene/2d/physics/shape_cast_2d.cpp ...
[ 11%] Compiling scene/2d/physics/static_body_2d.cpp ...
[ 11%] Compiling scene/2d/physics/touch_screen_button.cpp ...
scons: *** [bin/obj/scene/2d/tile_map_layer.linuxbsd.template_debug.x86_64.o] Error 1
scons: building terminated because of errors.
INFO: Time elapsed: 00:00:08.12
```
</details>

### Steps to reproduce

Try to build with Navigation 2D disabled:

```bash
# target=template_debug: Navigation 2D can only be disabled for export templates
# modules_enabled_by_default=no: to speed up building
scons target=template_debug disable_navigation_2d=yes modules_enabled_by_default=no
```

### Minimal reproduction project (MRP)

N/A

--- a/scene/2d/tile_map_layer.cpp
+++ b/scene/2d/tile_map_layer.cpp
@@ -1266,16 +1266,16 @@ void TileMapLayer::_physics_draw_quadrant_debug(const RID &p_canvas_item, DebugQ
 
 			if (face_index_array.size() > 2) {
 				Array face_mesh_array;
-				face_mesh_array.resize(Mesh::ARRAY_MAX);
-				face_mesh_array[Mesh::ARRAY_VERTEX] = Vector<Vector2>(face_vertex_array);
-				face_mesh_array[Mesh::ARRAY_INDEX] = Vector<int32_t>(face_index_array);
-				face_mesh_array[Mesh::ARRAY_COLOR] = Vector<Color>(face_color_array);
+				face_mesh_array.resize(RS::ARRAY_MAX);
+				face_mesh_array[RS::ARRAY_VERTEX] = Vector<Vector2>(face_vertex_array);
+				face_mesh_array[RS::ARRAY_INDEX] = Vector<int32_t>(face_index_array);
+				face_mesh_array[RS::ARRAY_COLOR] = Vector<Color>(face_color_array);
 				rs->mesh_add_surface_from_arrays(r_debug_quadrant.physics_mesh, RS::PRIMITIVE_TRIANGLES, face_mesh_array, Array(), Dictionary(), RS::ARRAY_FLAG_USE_2D_VERTICES);
 
 				Array line_mesh_array;
-				line_mesh_array.resize(Mesh::ARRAY_MAX);
-				line_mesh_array[Mesh::ARRAY_VERTEX] = Vector<Vector2>(line_vertex_array);
-				line_mesh_array[Mesh::ARRAY_COLOR] = Vector<Color>(line_color_array);
+				line_mesh_array.resize(RS::ARRAY_MAX);
+				line_mesh_array[RS::ARRAY_VERTEX] = Vector<Vector2>(line_vertex_array);
+				line_mesh_array[RS::ARRAY_COLOR] = Vector<Color>(line_color_array);
 
 				rs->mesh_add_surface_from_arrays(r_debug_quadrant.physics_mesh, RS::PRIMITIVE_LINES, line_mesh_array, Array(), Dictionary(), RS::ARRAY_FLAG_USE_2D_VERTICES);
 			}

